File: app-server/src/main/java/com/github/arhor/simple/expense/tracker/data/model/ExpenseItem.java
package com.github.arhor.simple.expense.tracker.data.model;

import lombok.Data;
import lombok.EqualsAndHashCode;

import java.math.BigDecimal;
import java.time.LocalDate;

import org.springframework.data.relational.core.mapping.Table;

@Data
@Table("expense_items")
@EqualsAndHashCode(callSuper = true)
public class ExpenseItem extends DomainObject<Long> {
    private LocalDate date;
    private BigDecimal amount;
    private String currency;
    private String comment;
    private Long expenseId;
}

File: app-server/src/main/kotlin/com/github/arhor/simple/expense/tracker/data/model/ExpenseItem.kt
package com.github.arhor.simple.expense.tracker.data.model

import org.springframework.data.annotation.Id
import org.springframework.data.annotation.Immutable
import org.springframework.data.relational.core.mapping.Table
import java.math.BigDecimal
import java.time.LocalDate

@Immutable
@Table("expense_items")
data class ExpenseItem(
    @Id
    val id: Long? = null,
    val date: LocalDate,
    val amount: BigDecimal,
    val currency: String,
    val comment: String? = null,
    val expenseId: Long,
)
